### ISSUE TITLE
Incremental Torch improvements

### ISSUE THREAD

**Issue by SergeyTsimfer** [analysiscenter/batchflow#457]
- refactor `BasePool`: it can be split into multiple classes with clear functionality

- improve `Encoder`/`Decoder` modules

- refactor `n_iters` and `decay` configurations: no need to pass `n_iters` in the root configuration

- make so every block is sent to `device`: can be helpful with pre-trained models

- refactor `pyramid` layers so they use common base

- make `Xception` out of `XceptionBlock`s

**Comment by nikita-klsh** [comment]
not specifying `layout` in the `body` of `TorchModel` config results in skipping `body` module at all

**Comment by SergeyTsimfer** [comment]
- add some warnings if task is `classification` and `classes` are not set in config: in this case, if model is built off of first batch with data, `classes` can be infered incorrectly (understandably so)

**Comment by SergeyTsimfer** [comment]
- make better name for `_add_output_op` in case of `callable` operation

-  allow `kwargs` for `TorchModel.save` method

**Comment by nikita-klsh** [comment]
- `kwargs`  for `combine` in `Decoder` do not reach the  `Combine` module inside.

**Comment by SergeyTsimfer** [comment]
At the time of implementation, we've decided to keep the argument names of `PyTorch` layers the same as their `TensorFlow` counterparts. For example, `dilation_rate` instead of `dilation`; `strides` instead of `stride`. I think it is time to fix that and either change names to the `PyTorch` convention or make methods recognize those parameters as aliases

**Comment by SergeyTsimfer** [comment]
Also, looks like we should change the letter of upsampling to `U` with parameter `mode`, and, maybe, `P` for pooling operations; otherwise, it is confusing to use `nn` interpolation which is also `n` letter (same as normalization layer)
